pkg/dispatcher/dispatcher.go

```diff
@@ -179,7 +180,11 @@ func Start() {
 			case event := <-dispatcher.eventChan:
 				switch v := event.(type) {
 				case events.ApplicationEvent:
-					getEventHandler(EventTypeApp)(v)
+					if v.GetEvent() == events.AppStateChange {
```

Comment:
I feel this could be misleading.
Can we define another interface for the app state change? In `pkg/common/events/events.go`, create another interface specifically for handling the app-state-updates from the core to the shim.

Reply:
Created a new interface: events.ApplicationStatusEvent